### ISSUE TITLE
Inconsistent behavior of __builtin__.round() for np.float64

### ISSUE THREAD

**Issue by moi90** [numpy/numpy#18610]
[`__builtin__.round()`](https://docs.python.org/3/library/functions.html#round) is supposed to return an `int` (unless ndigits is supplied):

> The return value is an integer if ndigits is omitted or None.

However, for `np.float64`, the result is `np.float64` again.

### Reproducing code example:

```python
import numpy as np

x = round(1.5)
x, type(x) # Gives (2, int)

x = round(np.float64(1.5))
x, type(x) # Gives (2.0, numpy.float64)
```

### NumPy/Python version information:

```
1.18.5 3.7.6 | packaged by conda-forge | (default, Jun  1 2020, 18:57:50) 
[GCC 7.5.0]
```

**Comment by IchiruTake** [comment]
> ```python
> import numpy as np
> 
> x = round(1.5)
> x, type(x) # Gives (2, int)
> 
> x = round(np.float64(1.5))
> x, type(x) # Gives (2.0, numpy.float64)
> ```

I don't think this was a bug. The rounding function is only access to the data and modify it based on binary approximation. This function won't create an 'int' object/entity and store data, either the 'ndigit' parameter is provided or None. If your wish come True, it would be hard to keep track of memory of non-deleting object (memory leakage, where object is created and deleted later without decreasing memory. Available when non-Python code like C/C++ access and affect Python code). 
Moreover, despite your thought, both 'int' and 'float' in Python comes from one parent object, while in Numpy (C mode), 'int' and 'float' is completely different architecture and representation. See Wikipedia, it shows how float and int are computed

```
import numpy as np

x = round(1.5)
print(x, type(x))

x = round(np.float64(1.5))
print(x, type(x))


x = round(2.6)
print(x, type(x))

x = round(np.float64(2.6))
print(x, type(x))

x = round(1.5)
print(x, type(x))

x = round(np.float32(1.5))
print(x, type(x))

x = round(2.6)
print(x, type(x))

x = round(np.float32(2.6))
print(x, type(x))

x = round(1.5)
print(x, type(x))

x = round(np.int8(1.5))
print(x, type(x))

x = round(2.6)
print(x, type(x))

x = round(np.int32(2.6))
print(x, type(x))

x = round(1.5)
print(x, type(x))

x = round(np.float64(1.5))
print(x, type(x))


x = round(1.4)
print(x, type(x))

x = round(np.float64(1.4))
print(x, type(x))

x = round(1.4)
print(x, type(x))

x = round(np.float32(1.4))
print(x, type(x))

x = round(1.4)
print(x, type(x))

x = round(np.float32(1.4))
print(x, type(x))

x = round(1.4)
print(x, type(x))

x = round(np.int8(1.4))
print(x, type(x))

x = round(1.4)
print(x, type(x))

x = round(np.int32(1.4))
print(x, type(x))
```

Link: https://docs.python.org/3/tutorial/floatingpoint.html#tut-fp-issues

**Comment by moi90** [comment]
@IchiruTake I'm sorry, I don't understand how anything of what you wrote has to do with this problem.

If it is not a bug, it was an active design decision to implement `np.float64.__round__` in a way that is incompatible with the standard implementation. And I doubt that memory issues play a role here.

Maybe someone overlooked that Python's `round` (`float` -> `int`) and C's `round` (`double` -> `double`) work differently?

**Comment by rkern** [comment]
It wasn't an oversight. Python's `round()` used to return `float`s as well. That changed in Python 3. We just have not changed the behavior of our floating point scalar types at the same time. It was a deliberate decision, but it's arguable that we ought to have made the change, too. I believe we decided to remain with the old behavior to match the rounding behavior of arrays, which can't change from `float64` to an arbitrary-sized integer array.

**Comment by rkern** [comment]
And this was indeed fixed in 1.19.0: https://numpy.org/doc/stable/release/1.19.0-notes.html#change-output-of-round-on-scalars-to-be-consistent-with-python

**Comment by moi90** [comment]
Thanks for investigating, @rkern ! I should have done that myself...
